src/error.c

```diff
@@ -160,6 +160,9 @@ static Obj FuncCURRENT_STATEMENT_LOCATION(Obj self, Obj context)
         (FIRST_EXPR_TNUM <= type && type <= LAST_EXPR_TNUM)) {
         Int line = LINE_STAT(call);
         Obj filename = GET_FILENAME_BODY(body);
+        if (!filename) {
+            filename = MakeImmString("<no filename>");
+        }
```

Comment:
Wouldn't it make more sense to check `filename` a few line above and then `return Fail;` as in all the other cases in this function that deal with missing location information?